gnucash/report/report-system/commodity-utilities.scm

```diff
@@ -129,8 +115,14 @@ construct with gnc:make-gnc-monetary and gnc:monetary->string instead.")
 ;;  (day2 23.33))
 ;; The intended second price is 6500/300, or 21.67.
 
-(define (gnc:get-commodity-totalavg-prices
-         currency-accounts end-date price-commodity report-currency)
+;; NOTE there is an optional #:commodity-splits argument which should
+;; contain a sorted splitlist with pricing information related to the
+;; price-commodity. See (gnc:get-commoditylist-totalavg-prices) how to
+;; generate this splitlist. This is NOT to be used by external
+;; reports.
+(define* (gnc:get-commodity-totalavg-prices
```

Comment:
If only the comment states this is NOT to be used by external reports, chances are someone will eventually ignore the warning and try to use the function in external reports anyway. Is there a way to prevent this in code also ?
Like splitting this up in a (define (gnc:get-commodity-totalavg-prices ...)) without the optional parameter and a, say (gnc:det-commodity-totalavg-prices-internal ...)) with the optional parameter. The former is exported, the latter isn't. The former will simply call the latter, and internal code can call the latter as well with optional parameter.

Reply:
More to the point, why shouldn't it be used by external reports if they can generate the splitlist?